cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaMirrorMakerCluster.java

```diff
@@ -0,0 +1,447 @@
+        KafkaMirrorMakerCluster kafkaMirrorMakerCluster = new KafkaMirrorMakerCluster(kafkaMirrorMaker.getMetadata().getNamespace(),
+                kafkaMirrorMaker.getMetadata().getName(),
+                Labels.fromResource(kafkaMirrorMaker).withKind(kafkaMirrorMaker.getKind()));
+
+        kafkaMirrorMakerCluster.setReplicas(kafkaMirrorMaker.getSpec() != null && kafkaMirrorMaker.getSpec().getReplicas() > 0 ? kafkaMirrorMaker.getSpec().getReplicas() : DEFAULT_REPLICAS);
+
+        if (kafkaMirrorMaker.getSpec() != null) {
+            kafkaMirrorMakerCluster.setWhitelist(kafkaMirrorMaker.getSpec().getWhitelist());
+            kafkaMirrorMakerCluster.setProducer(kafkaMirrorMaker.getSpec().getProducer());
+            kafkaMirrorMakerCluster.setConsumer(kafkaMirrorMaker.getSpec().getConsumer());
+            String image = kafkaMirrorMaker.getSpec().getImage();
+            kafkaMirrorMakerCluster.setImage(image == null ? KafkaMirrorMakerSpec.DEFAULT_IMAGE : image);
+            kafkaMirrorMakerCluster.setLogging(kafkaMirrorMaker.getSpec().getLogging());
+
+            Map<String, Object> metrics = kafkaMirrorMaker.getSpec().getMetrics();
+            if (metrics != null && !metrics.isEmpty()) {
+                kafkaMirrorMakerCluster.setMetricsEnabled(true);
+                kafkaMirrorMakerCluster.setMetricsConfig(metrics.entrySet());
+            }
+        }
+
+        if (kafkaMirrorMaker.getSpec().getConsumer() != null && kafkaMirrorMaker.getSpec().getConsumer().getAuthentication() != null) {
+            if (kafkaMirrorMaker.getSpec().getConsumer().getAuthentication() instanceof KafkaMirrorMakerAuthenticationTls) {
+                KafkaMirrorMakerAuthenticationTls consumerAuth = (KafkaMirrorMakerAuthenticationTls) kafkaMirrorMaker.getSpec().getConsumer().getAuthentication();
+                CertAndKeySecretSource consumerCertificateAndKey = consumerAuth.getCertificateAndKey();
+                if (consumerCertificateAndKey != null) {
+                    if (kafkaMirrorMaker.getSpec().getConsumer().getTls() == null) {
+                        log.warn("TLS configuration missing: related TLS client authentication will not work properly");
+                    }
+                    kafkaMirrorMakerCluster.setConsumerTlsAuthCertAndKey(consumerCertificateAndKey);
+                } else {
+                    log.warn("TLS Client authentication selected, but no certificate and key configured.");
+                    throw new InvalidResourceException("TLS Client authentication selected, but no certificate and key configured.");
+                }
+            } else if (kafkaMirrorMaker.getSpec().getConsumer().getAuthentication() instanceof KafkaMirrorMakerAuthenticationScramSha512)    {
+                KafkaMirrorMakerAuthenticationScramSha512 auth = (KafkaMirrorMakerAuthenticationScramSha512) kafkaMirrorMaker.getSpec().getConsumer().getAuthentication();
```

Comment:
Likewise I think you should be able to replace this with a method, and again use that new method for the producer.

Doing these things might be enough to drop the need to add this class to the checkstyle suppressions.